File: problems/1428/solution.py
# """
# This is BinaryMatrix's API interface.
# You should not implement it, or speculate about its implementation
# """
#class BinaryMatrix(object):
#    def get(self, x: int, y: int) -> int:
#    def dimensions(self) -> list[]:

class Solution:
    def leftMostColumnWithOne(self, binaryMatrix: 'BinaryMatrix') -> int:
        [rows, cols] = binaryMatrix.dimensions()
        
        rows -= 1
        cols -= 1
        
        row, col = 0, cols
        
        while row <= rows and col >= 0:
            if binaryMatrix.get(row, col):
                col -= 1
            else:
                row += 1
        
        return -1 if col == cols else col + 1

File: problems/1428/solution.cpp
/**
 * // This is the BinaryMatrix's API interface.
 * // You should not implement it, or speculate about its implementation
 * class BinaryMatrix {
 *   public:
 *     int get(int row, int col);
 *     vector<int> dimensions();
 * };
 */

class Solution {
public:
    int leftMostColumnWithOne(BinaryMatrix &binaryMatrix) {
        int rows = binaryMatrix.dimensions()[0] - 1,
            cols = binaryMatrix.dimensions()[1] - 1,
            row = 0,
            col = cols;
        
        while (row <= rows and col >= 0) {
            if (binaryMatrix.get(row, col) == 1) {
                col--;
            } else {
                row++;
            }
        }
        
        if (col >= cols) {
            return -1;
        }
        
        return col + 1;
    }
};